<a href="https://colab.research.google.com/github/JourdynLuv/JourdynLuv/blob/main/Razai_Khalid_1_5_Intro_to_Python_and_Pandas_Lab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lab Instructions

Find a dataset that interests you. I'd recommend starting on [Kaggle](https://www.kaggle.com/). Read through all of the material about the dataset and download a .CSV file.

1. Write a short summary of the data.  Where did it come from?  How was it collected?  What are the features in the data?  Why is this dataset interesting to you?  

2. Identify 5 interesting questions about your data that you can answer using Pandas methods.  

3. Answer those questions!  You may use any method you want (including LLMs) to help you write your code; however, you should use Pandas to find the answers.  LLMs will not always write code in this way without specific instruction.  

4. Write the answer to your question in a text box underneath the code you used to calculate the answer.



1. The data comes from a public database containing hundreds of pokemon across a couple generations, it includes the name of them, the types they are, legendary or not. It also includes stats such as HP, ATK, DEF, and more, the reason it is interesting to me is because I really enjoy Pokemon and Data so combining the two is my favorite.

2. Interesting Questions

1  What are the top 5 Pokemon with highest Base Attack?

2 How many pokemon belong to each Type 1 Category?

3 What is the average speed of Legendary Pokemon compared to all non legendary Pokemon?

4 Which Pokemon has the Absolute lowest HP in the game?

5 What are the names of all Pokemon that are both Dragon and Flying type?

In [2]:
import pandas as pd

url = 'https://gist.githubusercontent.com/armgilles/194bcff35001e7eb53a2a8b441e8b2c6/raw/92200bc0a673d5ce2110aaad4544ed6c4010f687/pokemon.csv'
df = pd.read_csv(url)

df.head()

top_attackers = df.sort_values(by='Attack', ascending=False).head(5)

top_attackers[['Name', 'Attack', 'Type 1', 'Legendary']]

,Name,Attack,Type 1,Legendary
163,MewtwoMega Mewtwo X,190,Psychic,True
232,HeracrossMega Heracross,185,Bug,False
429,DeoxysAttack Forme,180,Psychic,True
426,RayquazaMega Rayquaza,180,Dragon,True
424,GroudonPrimal Groudon,180,Ground,True


The 5 Pokemon with the top attack are
Mega Mewtwo X - 190
Mega Heracross - 185
Deoxys Attack Forme - 180
Mega Rayquaza - 180
Primal Groudon - 180

In [3]:
type_counts = df['Type 1'].value_counts()

type_counts

,count
Type 1,
Water,112
Normal,98
Grass,70
Bug,69
Psychic,57
Fire,52
Rock,44
Electric,44
Ground,32


The amount that belong to each Type 1 category is

Water - 112
Normal - 98
Grass - 70
Bug - 69
Psychic - 57
Fire - 52
Rock - 44
Electric - 44
Ground - 32
Ghost - 32
Dragon - 32
Dark - 31
Poison - 28
Fighting - 27
Steel - 27
Ice - 24
Fairy - 17
Flying - 4

In [4]:
speed_comparison = df.groupby('Legendary')['Speed'].mean()

speed_comparison

,Speed
Legendary,
False,65.455782
True,100.184615


The average speed of legendary pokemon is 100.18
while the average of non legendary is 65.45
with a difference of 34.73

In [5]:
lowest_hp = df.sort_values(by='HP').head(1)

lowest_hp[['Name', 'HP', 'Type 1', 'Type 2']]

,Name,HP,Type 1,Type 2
316,Shedinja,1,Bug,Ghost


Shedinja is the Pokemon with the lowest HP at 1.

In [6]:
dragon_flying = df[(df['Type 1'] == 'Dragon') & (df['Type 2'] == 'Flying')]

dragon_flying[['Name', 'Type 1', 'Type 2']]

,Name,Type 1,Type 2
161,Dragonite,Dragon,Flying
365,Altaria,Dragon,Flying
408,Salamence,Dragon,Flying
409,SalamenceMega Salamence,Dragon,Flying
425,Rayquaza,Dragon,Flying
426,RayquazaMega Rayquaza,Dragon,Flying


The Pokemon with Dragon and Flting typing is

Dragonite
Altaria
Salamence
Mega Salamence
Rayquaza
Mega Rayquaza